File: pset6/vigenere/vigenere.py
from sys import argv
from cs50 import get_string

# check no of args
if len(argv) is not 2:
    print("usage: python vigenere.py <key>")
    exit(1)

# check if all are alpha in key
for i in range(len(argv[1])):
    if not argv[1][i].isalpha():
        print("usage: python vigenere.py <alpha key>")
        exit(1)

# prompt for plain text
p = get_string("plaintext: ")
# list to hold cyphertext
ct = []
counter = 0
m = len(argv[1])
for i in range(len(p)):
    if p[i].isalpha():
        # to keep track of char in key
        j = counter % m
        # add key to plain text
        key = ord(argv[1][j].lower()) - ord('a')
        # j not > len(key) & j will not increase if j is greater than that
        if j is (m - 1):
            counter = 0
        else:
            counter += 1
        if p[i].isupper():
            a = ord(p[i])
            a -= 65
            temp = (a + key) % 26 + 65
            ct.append(chr(temp))
        elif p[i].islower():
            b = ord(p[i])
            b -= 97
            tmp = (b + key) % 26 + 97
            ct.append(chr(tmp))
    else:
        ct.append(p[i])
print("ciphertext: ", ''.join(ct))

File: pset2/vigenere/vigenere.c
#include <stdio.h>
#include <stdlib.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
//main should accept cmd line argument
int main(int argc, string argv[])
{
    //check if no. of args are 2
    if (argc != 2)
    {
        //print error if args are not equal to 2
        printf("error: please input a valid key in single argument after ./vigenere(alphabatic)\n");
        return 1;
    }
    //iterate the loop upto the length of the string argument 1 i.e, key and check if its all alphabatical
    for (int i = 0; i < strlen(argv[1]); i++)
    {
        if (! isalpha(argv[1][i]))
        {
            printf("error: input an alphabatic key\n");
            return 1;
        }
    }
    //get the plaintext input from user
    string p = get_string("plaintext: ");
    //declare a variable to store ciphertext of data type string with same no. of characters as in plaintext
    int n = strlen(p);
    int m = strlen(argv[1]);
    char c[n];
    int j;
    int counter = 0;
    printf("ciphertext: ");
    for (int i = 0; i < n; i++)
    {
        if (isalpha(p[i]))
        {
            //to keep track of characters in key use j
            j = counter % m;
            //to encipher we neeed to add key to the plaintext character
            int key = tolower(argv[1][j]) - 'a';
            /*-- j should not be graeater than no of elements in key
                 and j should not increase if j is greater than that --*/
            j == (m - 1) ? counter = 0 : counter++;
            if (isupper(p[i]))
            {
                p[i] -= 65;
                c[i] = (p[i] + key) % 26;
                c[i] += 65;
                printf("%c", c[i]);
            }
            else if (islower(p[i]))
            {
                p[i] -= 97;
                c[i] = (p[i] + key) % 26;
                c[i] += 97;
                printf("%c", c[i]);
            }

        }
        else
        {
            c[i] = p[i];
            printf("%c", c[i]);
        }
    }
    printf("\n");
    return 0;
}